# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/H264.h
# Commit: a14b6849e7b7
# Full Hash: a14b6849e7b71c8ef260f8c4baacb7d9d52e6b3e
# Author: alwu <alwu@mozilla.com>
# Date: 2019-09-07 09:40:21
# Description:
#   Bug 1579136 - do not read content from null NAL. r=jya
#   
#   Dealing with a case where `DecodeNALUnit()` returns nullptr which we should not try to create bit reader and decode anything.
#   
#   In addition, wrap these logic to `DecodedISlice()` in order to keep consistent with a case where we decode SEI.
# ==============================================================================

diff -r 3f813fd91f92 -r a14b6849e7b7 dom/media/platforms/agnostic/bytestreams/H264.h
--- a/dom/media/platforms/agnostic/bytestreams/H264.h	Fri Sep 06 20:08:52 2019 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/H264.h	Fri Sep 06 20:11:22 2019 +0000
@@ -510,6 +510,9 @@
   // point.
   static bool DecodeRecoverySEI(const mozilla::MediaByteBuffer* aSEI,
                                 SEIRecoveryData& aDest);
+  // Decode NAL Slice payload and return true if its slice type is I slice or SI
+  // slice.
+  static bool DecodeISlice(const mozilla::MediaByteBuffer* aSlice);
 };
 
 }  // namespace mozilla
